### PR TITLE
release docs updated

### LINKED ISSUE #865: Document and perform a Keycloak Benchmark Release
Follow-up: document for users which version to pick for their local tests.

### PR DESCRIPTION
**Pull request by kami619**
fixes #865

### CODE REVIEW

**Review by ryanemerson — COMMENTED** [review]
Should we document what version(s) of Keycloak a branch is compatible with in the README file?

File: doc/benchmark/modules/ROOT/pages/release.adoc [hunk 1]
```diff
@@ -1,53 +1,47 @@
...
-git checkout keycloak-benchmark-parent-0.1
-mvn clean install
-git checkout main
-----
-+
-Check that this file (corresponding to your released version) exists on your laptop: `<YOUR_HOME>/.m2/repository/org/keycloak/keycloak-benchmark-dataset/0.1/keycloak-benchmark-dataset-0.1.jar`
-
-If everything is OK, a GitHub release is going to be created automatically with all artifacts that should be available for download.
-
+We would release a new version of Keycloak Benchmark
```
[the remark this answers is not in the document]
Comment (kami619) [reply doc/benchmark/modules/ROOT/pages/release.adoc:3]: I was being more assertive, but we could err on the side of caution and be `aiming` to release a version.

[the remark this answers is not in the document]
Comment (ryanemerson) [reply doc/benchmark/modules/ROOT/pages/release.adoc:3]: I'm fine with being more assertive, but in that case I think it should be "We will release a new version ..."

**Comment by kami619** [comment]
> Should we document what version(s) of Keycloak a branch is compatible with in the README file?

Did you mean inside this doc https://github.com/keycloak/keycloak-benchmark/blob/0.13-release/README.adoc for that specific release? We are doing that on the release artifact page currently - https://github.com/keycloak/keycloak-benchmark/releases

File: doc/benchmark/modules/ROOT/pages/release.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (kami619) [reply doc/benchmark/modules/ROOT/pages/release.adoc:3]: Agreed, I will change it to `We will release a new version ...`

**Comment by ryanemerson** [comment]
I was thinking in the repositories [README.adoc](https://github.com/keycloak/keycloak-benchmark/blob/main/README.adoc) so that users can quickly see whether `main` is applicable to their Keycloak version.

Do we document anywhere that users should use branch `xx-release` for specific Keycloak versions? I think this would also be nice to have in the README so that users don't try to execute `main` against an older version etc.

**Comment by kami619** [comment]
> I was thinking in the repositories [README.adoc](https://github.com/keycloak/keycloak-benchmark/blob/main/README.adoc) so that users can quickly see whether `main` is applicable to their Keycloak version.
> 
> Do we document anywhere that users should use branch `xx-release` for specific Keycloak versions? I think this would also be nice to have in the README so that users don't try to execute `main` against an older version etc.

Good point, let me think of a way to represent this in the README.adoc of the specific branches, including `main`.

**Comment by kami619** [comment]
> I was thinking in the repositories [README.adoc](https://github.com/keycloak/keycloak-benchmark/blob/main/README.adoc) so that users can quickly see whether `main` is applicable to their Keycloak version.
> 
> Do we document anywhere that users should use branch `xx-release` for specific Keycloak versions? I think this would also be nice to have in the README so that users don't try to execute `main` against an older version etc.

How about we add this to the [README.adoc](https://github.com/keycloak/keycloak-benchmark/blob/main/README.adoc) in the main branch? And I can add a comment to update this file as well when a release branch is cut. One task would be to retroactively add this block to all the older release branches.

```
== Keycloak Compatibility
This `main` branch Keycloak benchmark works with the Keycloak nightly versions, for specific Keycloak release-based binaries please look at https://github.com/keycloak/keycloak-benchmark/releases.```

**Comment by ryanemerson** [comment]
That looks great :+1: 

> One task would be to retroactively add this block to all the older release branches.

IMO most people will read the README on `main` and then navigate to the required branch from there, so I think we can skip this effort.

**Review by ahus1 — APPROVED** [review]
@kami619 - I reviewed the docs, and it looks great, thank you for detailing all the steps. One thing I did is that I used the `{github-files}` which will in the rendered version of the docs point to the GitHub repo.

The new docs show where we should remove the need to edit those files by retrieving the values for example from the main `pom.xml` file to avoid editing those files at all. This would be a follow-up task. 

I leave this open in case you want to add a note to the `README.adoc`. IMHO it is ok add it only to main. A second place where to add it would be `doc/ROOT/modules/ROOT/pages/index.adoc`.

**Pull request merged**